src/casts/convert-closures.rkt

```diff
@@ -1,551 +1,2315 @@
+              (car not-wk-c*))
+
+            (when (or (eq? fst-mode 'code-only)
+                      (eq? fst-mode 'closure-only))
+              (unless (null? (cdr not-wk-c*))
+                (error 'todo)))
+                        
+            (hash-set! label->closure-status fst-label 'regular)
+            
+            (define rst-nwk* (cdr not-wk-c*))
+
+
+            (define-values (fst-fv-set uid->uid^)
+              (resolve-free-variables eliminated? uid->uid fst-fv-ls))
+
+            #;
+            (define fst-fv-set
+              (for/fold ([fv-set : (Setof Uid) (seteq)])
+                        ([fv fst-fv-ls] #:when (not (eliminated? fv)))
+                (set-add fv-set (or (hash-ref uid->uid fv #f) fv))))
+
+            ;; Add fvs of wk closure to fvs of
+            ;; - don't add fst-name if present
+            ;; TODO use best uid->uid here 
+            (define-values (fv-set/wk-names/fst-name _)
+              (for*/fold ([fv-set : (Setof Uid) fst-fv-set]
+                          [u->u : (HashTable Uid Uid) uid->uid])
+                         ([c wk-c*])
+                (resolve-free-variables eliminated? u->u
+                                        (Closure-free-vars c)
+                                        fv-set)))
+
+            (define fv-set/wk-names
+              (set-remove fv-set/wk-names/fst-name fst-name))
+            
+            (define fv-set (set-subtract fv-set/wk-names wk-name-set))
+            
+            (define fv-ls (sort (set->list fv-set) uid<?))
+
+            ;; Have to figure out sharable closures before recurring
+            (define fst-ext-sc (shared-closure fst-name fv-ls))
+            (define fst-int-sc (shared-closure fst-self fv-ls))
+            (hash-set! uid->fv-set.clos fst-name (cons fv-set fst-ext-sc))
+            (hash-set! uid->fv-set.clos fst-self (cons fv-set fst-int-sc))
+
+            ;; In the other !wk closures and the body the
+            ;; well-known closure just use the fst closure
+            (define uid->uid^^ (map-uid*-to uid->uid^ wk-name* fst-name))
+
+            
+            (define-values (rst-fv-ls* rst-fv-set* rst-int-sc* rst-ext-sc*)
+              ;; TODO consider threading the uid map through
+              (for/lists ([fv-ls* : (Listof Uid*)]
+                          [fv-set* : (Listof (Setof Uid))]
+                          [int-sc* : (Listof Shared-Closure)]
+                          [ext-sc* : (Listof Shared-Closure)])
+                         ([c rst-nwk*])
+                (match-define (Closure name #f 'regular label self
+                                       ctr? f* p* rec-e)
+                  c)
+                (hash-set! label->closure-status label 'regular)
+                (define-values (fv-set _) 
+                  (resolve-free-variables eliminated? uid->uid^^ f*)
+                  #;
+                  (for*/fold ([fv-set : (Setof Uid) (seteq)])
```

Comment:
please delete commented code.

Reply:
Currently yes. Though I am starting to think we should just remove data proxies, and only support the hybrid model.